pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -134,16 +136,23 @@ def test_combine_first_mixed_bug(self):
         expected = DataFrame({"A": [1, 2, 3, 5, 3, 7.0], "B": [np.nan, 2, 3, 4, 6, 8]})
         tm.assert_frame_equal(result, expected)
 
+    def test_combine_first_return_obj_type_with_bools(self):
         # GH3552, return object dtype with bools
         df1 = DataFrame(
             [[np.nan, 3.0, True], [-4.6, np.nan, True], [np.nan, 7.0, False]]
         )
         df2 = DataFrame([[-42.6, np.nan, True], [-5.0, 1.6, False]], index=[1, 2])
 
-        result = df1.combine_first(df2)[2]
-        expected = Series([True, True, False], name=2)
-        tm.assert_series_equal(result, expected)
+        expected1 = pd.Series([True, True, False], name=2, dtype=object)
```

Comment:
can you parameterize this test (e.g. put the df1 and df2 in the parameter along with the expected)